src/save_processes.c

```diff
@@ -0,0 +1,540 @@
+    main->tilemapUpdateBits = 0xC;
+    gLCDIORegisters.lcd_bg2cnt = BGCNT_PRIORITY(1) | BGCNT_CHARBASE(0) | BGCNT_SCREENBASE(30) | BGCNT_16COLOR | BGCNT_WRAP | BGCNT_TXT256x256;
+    gLCDIORegisters.lcd_bg3vofs = 8;
+    gLCDIORegisters.lcd_bg3hofs = 8;
+    gLCDIORegisters.lcd_bg1vofs = 0;
+    gLCDIORegisters.lcd_bg1hofs = 0;
+    main->showTextboxCharacters = FALSE;
+    StartHardwareBlend(1, 0, 1, 0x1F);
+    main->process[GAME_SUBPROCESS]++;
+}
+
+void SaveGameInitButtonsSubProcess(struct Main *main)
+{
+    struct OamAttrs * oam;
+    sub_8002878(&gCourtRecord);
+    if(gCourtRecord.unk1 == 0) // ?
+    {
+        main->advanceScriptContext = TRUE;
+        main->showTextboxCharacters = TRUE;
+        gScriptContext.currentSection = 0xFFFF;
+        if(main->sIsEpisodePartOver)
+            ChangeScriptSection(0);
+        else
+            ChangeScriptSection(1);
+        gScriptContext.textXOffset = 9;
+        gScriptContext.textYOffset = 52;
+        oam = &gOamObjects[40];
+        oam->attr0 = SPRITE_ATTR0(96, ST_OAM_AFFINE_OFF, ST_OAM_OBJ_BLEND, FALSE, ST_OAM_4BPP, ST_OAM_H_RECTANGLE);
+        oam->attr1 = SPRITE_ATTR1_NONAFFINE(48, FALSE, FALSE, 3);
+        oam->attr2 = SPRITE_ATTR2(0x1E0, 0, 10);
+        oam++;
+        oam->attr0 = SPRITE_ATTR0(96, ST_OAM_AFFINE_OFF, ST_OAM_OBJ_BLEND, FALSE, ST_OAM_4BPP, ST_OAM_H_RECTANGLE);
+        oam->attr1 = SPRITE_ATTR1_NONAFFINE(128, FALSE, FALSE, 3);
+        oam->attr2 = SPRITE_ATTR2(0x200, 0, 10);
+        main->blendCounter = 0;
+        main->blendDelay = 1;
+        main->blendDeltaY = 0x10;
+        gLCDIORegisters.lcd_bldcnt = BLDCNT_EFFECT_BLEND | BLDCNT_TGT2_BG3;
+        gLCDIORegisters.lcd_bldalpha = BLDALPHA_BLEND(0, main->blendDeltaY);
+        if(main->sIsEpisodePartOver)
+            main->selectedButton = 0;
+        else
+            main->selectedButton = 1;
+        main->process[GAME_SUBPROCESS]++;
+    }
+}
+
+void SaveGameWaitForInputSubProcess(struct Main *main)
+{
+    struct OamAttrs * oam;
+    if(gScriptContext.unk0 & 0x8)
+    {
+        if(gJoypad.pressedKeysRaw & (DPAD_RIGHT | DPAD_LEFT))
+        {
+            main->selectedButton ^= 1;
+            PlaySE(0x2A);
+        }
+        else if(gJoypad.pressedKeysRaw & A_BUTTON)
+        {
+            PlaySE(0x40);
+            if(main->selectedButton == 0)
+            {
+                if(main->sIsEpisodePartOver)
+                    gSaveDataBuffer.main.unk17 &= 0xF0;
+                else
+                    gSaveDataBuffer.main.unk17 |= 0x1;
+                if(SaveGameData())
+                {
+                    gScriptContext.currentSection = 0xFFFF;
+                    ChangeScriptSection(3);
+                    gScriptContext.textXOffset = 9;
+                    gScriptContext.textYOffset = 52;
+                    main->process[GAME_SUBPROCESS] = 6;
+                    return;
+                }
+                if(main->sIsEpisodePartOver == FALSE)
```

Comment:
`if(!main->sIsEpisodePartOver)`

(etc for other explicit comparisons to this pseudo field)

Reply:
addresses